cpp/test/packed/packed_integration_test.cpp

```diff
@@ -38,7 +38,28 @@ class PackedIntegrationTest : public PackedTestBase {
   PackedIntegrationTest() : props_(*parquet::default_writer_properties()) {}
 
   void SetUp() override {
-    ASSERT_AND_ASSIGN(fs_, BuildFileSystem("file:///tmp/", &file_path_));
+    const char* access_key = std::getenv("ACCESS_KEY");
+    const char* secret_key = std::getenv("SECRET_KEY");
+    const char* endpoint_url = std::getenv("S3_ENDPOINT_URL");
+    if (access_key == nullptr || secret_key == nullptr || endpoint_url == nullptr) {
+      ASSERT_AND_ASSIGN(fs_, BuildFileSystem("file:///tmp/", &file_path_));
+    } else {
+      arrow::fs::S3GlobalOptions global_options;
+      auto status = arrow::fs::InitializeS3(global_options);
+      EXPECT_TRUE(status.ok());
+      std::atexit([]() {
+          auto status = arrow::fs::EnsureS3Finalized();
+          if (!status.ok()) {
+            throw std::runtime_error("Failed to finalize S3: " + status.message());
+          }
+        });
+      arrow::fs::S3Options options;
+      file_path_ = "30k_4MB_write_and_read";
```

Comment:
`file_path_` can be a local variable.

PS, is it necessary to delete the files when tearing down the test?